### PR TITLE
adds random station rooms!

### PR DESCRIPTION
**Pull request by Amelia0010**
### Intent of your Pull Request

https://github.com/yogstation13/yogstation/pull/2122, except they're called station rooms and they can add templates within a template :) 

I intend to randomize maint somewhat more than prebase, so we can have randomly chosen maint segments and possibly station segments, but for now this is just a framework.

I included a .dmm file which is that surgery thing in arrivals, just to make sure it works :)

### CODE REVIEW

**Comment by Kmc2000** [comment]
Cool!

File: code/_globalvars/lists/mapping.dm [hunk 1]
```diff
@@ -6,6 +6,7 @@ GLOBAL_LIST_EMPTY(landmarks_list)				//list of all landmarks created
 GLOBAL_LIST_EMPTY(start_landmarks_list)			//list of all spawn points created
 GLOBAL_LIST_EMPTY(department_security_spawns)	//list of all department security spawns
 GLOBAL_LIST_EMPTY(generic_event_spawns)			//list of all spawns for events
+GLOBAL_LIST_EMPTY(stationroom_landmarks)		//yogs - list of all spawns for stationrooms
```
Comment (Kmc2000) [inline code/_globalvars/lists/mapping.dm:9]: MoDulAr CoDe

File: yogstation/code/controllers/subsystem/mapping.dm [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/datum/controller/subsystem/mapping
+	var/list/station_room_templates = list()
+
+/datum/controller/subsystem/mapping/proc/seedStation()
+	for(var/V in GLOB.stationroom_landmarks)
+		var/obj/effect/landmark/stationroom/LM = V
+		LM.load()
+
+	if(GLOB.stationroom_landmarks.len)
+		seedStation() //I'm sure we can trust everyone not to insert a 1x1 rooms which loads a landmark which loads a landmark which loads a la...
```
Comment (Kmc2000) [inline yogstation/code/controllers/subsystem/mapping.dm:10]: Think again

File: yogstation/code/datums/ruins/station.dm [hunk 3]
```diff
@@ -0,0 +1,8 @@
+/datum/map_template/ruin/station
+	prefix = "_maps/RandomRuins/StationRuins/"
+	cost = 0
```
Comment (Kmc2000) [inline yogstation/code/datums/ruins/station.dm:3]: Careful that the ruinloader doesn't pick these up? Or have you accounted for that

**Review by Kmc2000 — COMMENTED** [review]
Seems fine, now add pictures of the ruins

File: yogstation/code/datums/ruins/station.dm (hunk 3 above)
Comment (Amelia0010) [reply yogstation/code/datums/ruins/station.dm:3]: ruinloader does space ruins and lava ruins specifically, not ruins in general :)

File: yogstation/code/controllers/subsystem/mapping.dm (hunk 2 above)
Comment (Amelia0010) [reply yogstation/code/controllers/subsystem/mapping.dm:10]: well if that happens, then maintainers are to blame not my PR :^)

File: code/_globalvars/lists/mapping.dm (hunk 1 above)
Comment (Amelia0010) [reply code/_globalvars/lists/mapping.dm:9]: 😕

**Comment by Nickvr628** [comment]
Lots of cool possibilities for this. I like it.

**Comment by Amelia0010** [comment]
@yogstation13/maintainers this is ready now :)

**Comment by Time-Green** [comment]
You could probably PR this to /tg/ aswell

**Comment by Amelia0010** [comment]
why on god's green earth would I do that?

**Comment by Time-Green** [comment]
You wouldn’t have to worry about keeping it seperate, and more people get to enjoy your features?

**Comment by Amelia0010** [comment]
I'm not hiding anything, and if I PR all my stuff to teegee then yogs'll never be less of a teegeeclone

**Comment by Time-Green** [comment]
Errr, I meant keeping it in the yog folder, not hiding. No idea what happened there

**Pull request merged**